centreon/src/Core/Common/Infrastructure/Repository/ApiCallIterator.php

```diff
@@ -101,14 +102,16 @@ public function count(): int
      * @throws ServerExceptionInterface
      * @throws RedirectionExceptionInterface
      * @throws ClientExceptionInterface
+     * @throws RepositoryException
      */
     private function createCache(): void
     {
         if (! $this->isCachedCreated) {
             $fromPage = 1;
             $totalPage = 0;
             do {
-                $url = $this->url . sprintf('?limit=%d&page=%d', $this->maxItemsByRequest, $fromPage);
+                $separator = isset(parse_url($this->url)['query']) ? '&' : '?';
```

Comment:
Suggestion, maybe simpler ?
```suggestion
                $separator = parse_url($this->url, PHP_URL_QUERY) ? '&' : '?';
```